### ISSUE TITLE
Provide an operator for Oopenshift/Kubernetes or contribute to external-secrets-operator (example)

### ISSUE THREAD

**Issue by flybyray** [Keeper-Security/secrets-manager-go#5]
Describe the solution you'd like
Keeper is a pretty popular secret manager and should provide some useful integration with platforms like openshift/kubernetes.

What is the added value?
Any organization who uses Keeper would be able to retain it as the single source of truth for secrets and not need to use another backend.

The current "integration" https://docs.keeper.io/secrets-manager/secrets-manager/integrations/kubernetes is somewhat incomplete.

> [Using Commander, create a Secrets Manager device configuration for Kubernetes. Note that this configuration is not IP locked and it is **pre-initialized**.](https://docs.keeper.io/secrets-manager/secrets-manager/integrations/kubernetes#create-a-secrets-manager-configuration)

This "pre-initialized" means something like a static token with consequences:

> Static token authentication is fine if you want quickly get started with Keeper Secrets Management in kubernetes, but a static token is not protected any further. Any disclosure to unintended parties allows Keeper Secrets use with the associated token roles.

From a community driven project you can read how a more complete integration could look like:

> [External Secrets Operator is a Kubernetes operator that integrates external secret management systems like AWS Secrets Manager, HashiCorp Vault, Google Secrets Manager, Azure Key Vault and many more. The operator reads information from external APIs and automatically injects the values into a Kubernetes Secret.](https://external-secrets.io/v0.4.4/)

Corresponding issue on a community driven project: https://github.com/external-secrets/external-secrets/issues/476
Maybe update roadmap description at https://docs.keeper.io/secrets-manager/secrets-manager/overview

**Comment by ppodevlab** [comment]
@flybyray https://external-secrets.io/main/provider/keeper-security/

**Comment by ppodevlab** [comment]
> @ppodevlab this is really cool integration. Thank you for working on it. Do you happen to know when it will be part of the release?

no, but it shouldn't take too much, as there has not been any release for a few weeks now.
